src/lib/components/withAuthorization/with-authorization.component.js

```diff
@@ -11,7 +10,8 @@ export const withAuthorization = (Component, Loader) =>
           case undefined:
             return Loader || null;
           case null:
-            return <Redirect to="/login" />;
+            window.location.href = '/login';
+            return null;
```

Comment:
Yes, and updating the version in this library's package.json file was an attempt to bring the generator and components lib to the same version. Unfortunately when I updated locally, it still did not fix the issue.

We also ran into some problems because the "right" way to do manual redirects is to use hooks, but we cannot use hooks in class components. This was the compromise implementation, and in my opinion isn't bad, as this will cause an app refresh on logout, which often helps reduce issues from left-over state anyway.

I can add a comment explaining all this into the code.

Reply:
Another update - we don't have a ticket for this debt piece yet as we don't have any true "debt" tickets, which will likely be the outcome of the audit we're doing right now.

Reply:
Well, since the debt is being introduced now, isn't this the time to create that ticket? (I'd use a GitHub Issue, btw, since everyone potentially reading the code should be able to refer to it or even take it upon themselves to fix it.)

I'd also prefer specifying what "didn't like `<Redirect>` means, e.g. what error message did you get, what behaviour was different from what you'd expect?

Btw, if a hook is the right way to do it, couldn't this component just be converted into a Functional Component? Doesn't seem to do much other than rendering?